authentik/policies/engine.py

```diff
@@ -130,24 +130,26 @@ def build(self) -> "PolicyEngine":
                 if self._check_cache(binding):
                     continue
                 self.logger.debug("P_ENG: Evaluating policy", binding=binding, request=self.request)
-                our_end, task_end = Pipe(False)
-                task = PolicyProcess(binding, self.request, task_end)
+
+                result_queue = get_context().Queue()
+                task = PolicyProcess(binding, self.request, result_queue)
                 task.daemon = False
+
                 self.logger.debug("P_ENG: Starting Process", binding=binding, request=self.request)
                 if not CURRENT_PROCESS._config.get("daemon"):
                     task.run()
                 else:
                     task.start()
                 self.__processes.append(
-                    PolicyProcessInfo(process=task, connection=our_end, binding=binding)
+                    PolicyProcessInfo(process=task, result_queue=result_queue, binding=binding)
                 )
             # If all policies are cached, we have an empty list here.
             for proc_info in self.__processes:
                 if proc_info.process.is_alive():
                     proc_info.process.join(proc_info.binding.timeout)
                 # Only call .recv() if no result is saved, otherwise we just deadlock here
                 if not proc_info.result:
-                    proc_info.result = proc_info.connection.recv()
+                    proc_info.result = proc_info.result_queue.get()
```

Comment:
Having played with this a bit, I think that this call might block forever if a task times out.

Reply:
Yes though it's possible for me to change this to also have its own timeout. Shall it just raise an exception in this case? And how long should the timeout be if so?

Reply:
I've gone ahead and made changes to this. Please let me know if the exception raised is appropriate for this circumstance.